src/firewall/core/nftables.py

```diff
@@ -238,17 +239,37 @@ def __init__(self, fw):
         self._nft_handle.set_handle_output(True)
         self._nft_handle.set_json_output(True)
 
-    def nft_cmd(self, json_root):
+    def nft_cmd(self, json_root, is_large=False):
         if isinstance(json_root, list):
             # The caller gave us the JSON in a list, so we can drop the
             # reference to the memory. Unpack the list, and delete the entry.
             l = json_root
             (json_root,) = l
             del l[0]
-        json_root_str = json.dumps(json_root)
-        del json_root
-        json_root_str = json_root_str.encode("utf-8")
-        rc, output, error = self._nft_handle.cmd(json_root_str)
+
+        if is_large and hasattr(self._nft_handle, "cmd_from_file"):
```

Comment:
This `hasattr()` guard is weird. AFAIK, python-nftables has always had this function.

Reply:
it's only "recently" added to the python library, here: https://git.netfilter.org/nftables/commit/?id=448bd104d3530ed2870921a35ebf54589cdb274c

This is recent enough, that github CI tests would break without this.